fdbclient/NativeAPI.actor.cpp

```diff
@@ -233,10 +252,18 @@ void DatabaseContext::updateCachedReadVersion(double t, Version v) {
 }
 
 Version DatabaseContext::getCachedReadVersion() {
+	if (sharedStatePtr) {
+		MutexHolder mutex(sharedStatePtr->mutexLock);
```

Comment:
I would suggest to plan a follow up task to optimize synchonization on the global cachedReadVersion. Mutex gives a reliable implementation, but it can introduce a significant overhead in a configuration with multiple FDB threads. I think we can replace mutex with atomics. It is not completely trivial, because the cachedReadVersion and lastGrvTime must be updated atomically, but it is doable. We can discuss it after you are done with this PR.